lib/sm/api/folders.rb

```diff
@@ -29,17 +29,17 @@ def delete_folder(id)
         response.nil? ? nil : response.status
       end
 
-      # FIXME: this is going to need better exception handling for multiple GET requests
-      # get_folder_messages:  Retrieves all messages
       def get_folder_messages(folder_id)
-        folder = get_folder(folder_id).attributes
-        page_count = (folder[:count] / MHV_MAXIMUM_PER_PAGE.to_f).ceil
+        page = 1
         json = { data: [], errors: {}, metadata: {} }
 
-        (1..page_count).each do |page|
+        loop do
           path = "folder/#{folder_id}/message/page/#{page}/pageSize/#{MHV_MAXIMUM_PER_PAGE}"
-          page_data = perform(:get, path, nil, token_headers)[:data]
-          json[:data].concat(page_data)
+          page_data = perform(:get, path, nil, token_headers)
+          json[:data].concat(page_data[:data])
+          json[:metadata].merge(page_data[:metadata])
+          break unless page_data[:data].size == MHV_MAXIMUM_PER_PAGE
```

Comment:
Just a nit pick - it there are a multiple of MHV_MAXIMUM_PER_PAGE messages in the folder, then you would be doing an extra call, again just a nit-pick. Otherwise LGTM.